@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,451 @@
+
+    if (!raycaster.ray.intersectsSphere(tmpSphere)) {
+        return;
+    }
+
+    for (let i = beginIdx; i < endIdx; i += SEGMENT_STRIDE) {
+        const a = indices[i];
+        const b = indices[i + VERTEX_STRIDE];
+
+        // Find the plane containing the line segment, using the segment start, end and extrusion
+        // vector.
+        line.start.fromBufferAttribute(position, a);
+        line.end.fromBufferAttribute(position, b);
+        vExt.set(bitangent.getX(a), bitangent.getY(a), bitangent.getZ(a)).normalize();
+        plane.setFromCoplanarPoints(line.start, tmpV3.copy(line.start).add(vExt), line.end);
+
+        // The ray intersection if any, will be on the extrusion plane.
+        if (!localRay.intersectPlane(plane, interPlane)) {
+            continue;
+        }
+
+        const interLine = tmpV3;
+        if (
+            !intersectLine(
+                localRay,
+                line,
+                vExt,
+                lHalfWidth,
+                lHalfWidthSq,
+                plane,
+                interPlane,
+                interLine
+            )
+        ) {
+            continue;
+        }
+
+        // Move back to world space for distance calculation
+        const interLineWorld = interLine.applyMatrix4(mesh.matrixWorld);
+
+        const distance = raycaster.ray.origin.distanceTo(interLineWorld);
+
+        if (distance < raycaster.near || distance > raycaster.far) {
+            continue;
+        }
+
+        intersects.push({
+            distance,
+            point: interLineWorld.clone(),
+            index: i,
+            object: mesh
+        });
+    }
+}
+
+const singleFeatureStart = [0];
+
+/**
+ * Finds the intersections of a ray with a group within a solid line mesh.
+ * @param mesh The mesh whose intersections will be found.
+ * @param material The material used by the group inside the mesh.
+ * @param raycaster  Contains the intersection ray.
+ * @param localRay Same ray as raycaster.ray but in object local space.
+ * @param intersects  Array that will contain all intersections found between ray and feature.
+ * @param firstFeatureIdx Index of the first feature in the group.
+ * @param groupEndIdx Index of the last vertex in the group.
+ * @returns The next feature index after the group.
+ */
+function intersectGroup(
+    mesh: THREE.Mesh,
+    material: THREE.Material,
+    raycaster: THREE.Raycaster,
+    localRay: THREE.Ray,
+    intersects: THREE.Intersection[],
+    firstFeatureIdx: number,
+    groupEndIdx: number
+): number {
+    const bVolumes = mesh.userData.feature.boundingVolumes;
+    const geometry = mesh.geometry as THREE.BufferGeometry;
+    assert(isSolidLineMaterial(material), "Unsupported material type");
+    const solidLineMaterial = material as SolidLineMaterial;
+
+    const halfWidth = (solidLineMaterial.lineWidth + solidLineMaterial.outlineWidth) / 2;
+    const localHalfWidth = halfWidth / ((mesh.scale.x + mesh.scale.y + mesh.scale.z) / 3);
```

Comment:
Shouldn't it be?
```
const localHalfWidth = halfWidth / mesh.scale.length()
```

Reply:
That's wrong, just see what would happen with no scaling (scale = [1,1,1]), the width would be divided by sqrt(3).

I took the formula above from THREE.js line picking. The result is only correct for uniform scaling (scale.x === scale.y === scale.z).  For non-uniform scaling we would need to compute the localHalfWidth depending on what direction is measured which would require more computations.
This here is an approximation assuming the scaling factor difference on each dimension is small enough. It holds now since we never scale at all (uniformly or not).
I'll add some clarification and an assert to catch any issue in the future.